### PR TITLE
doc: add blobfuse mount with managed identity doc

### PR DESCRIPTION
**Pull request by umagnus**
**What type of PR is this?**

/kind documentation

**What this PR does / why we need it**:

doc: add mount azure blob storage doc
referring to https://github.com/qxsch/Azure-Aks/tree/master/aks-blobfuse-mi

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: deploy/example/mountstorage/README.md [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+        --account-name myaksblob \
+        --container-name mycontainer \
+        --name test.htm \
+        --file test.htm \
+        --auth-mode key --account-key "$(az storage account keys list --account-name "$storageaccountname" --query '[0].value' -o tsv)"
+    ```
+
+## Using kubelet identity
+
+1. Give kubelet identity access to storage account
```
Comment (andyzhangx) [inline deploy/example/mountstorage/README.md:34]: option#1: grant kubelet identity access to storage account

File: deploy/example/mountstorage/README.md [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+    said="$(az storage account list -g "$resourcegroup" --query "[?name == '$storageaccountname'].id" -o tsv)"
+    az role assignment create --assignee-object-id "$kloid" --role "Storage Blob Data Owner" --scope "$said"
+    ```
+
+1. Get the clientID of kubelet identity
+    ```bash
+    az identity list -g "$resourcegroup" --query "[?name == 'aks-fuseblob-mi-agentpool'].clientId" -o tsv
+    ```
+
+## Using a dedicated user-assigned managed indentity
```
Comment (andyzhangx) [inline deploy/example/mountstorage/README.md]: option#2:

File: deploy/example/mountstorage/README.md [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+- Create a storage account container and upload file, e.g.
+    ```bash
+    resourcegroup="aks-fuseblob-mi"
+    storageaccountname="myaksblob"
+    az storage account create -g "$resourcegroup" -n "$storageaccountname" --access-tier Hot  --sku Standard_LRS
+    az storage container create -n mycontainer --account-name "$storageaccountname" --public-access off
+    az storage blob upload \
+        --account-name myaksblob \
+        --container-name mycontainer \
+        --name test.htm \
```
Comment (andyzhangx) [inline deploy/example/mountstorage/README.md:27]: let's remove this application example, make it simple

File: deploy/example/mountstorage/README.md [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+          volumes: 
+            - name: pvc-blob1 
+              persistentVolumeClaim: 
+                claimName:  pvc-blob1
+    status: {}
+    
+    ---
+    
+    apiVersion: v1
+    kind: Service
```
Comment (andyzhangx) [inline deploy/example/mountstorage/README.md:157]: let's remove this example, make it simple

File: deploy/example/mountstorage/README.md [hunk 5]
```diff
@@ -0,0 +1,276 @@
...
+    # create pv and pvc
+    kubectl apply -f volume2.yaml
+    # check it
+    kubectl get pv
+    kubectl get pvc
+    ```
+
+1. Now you can use the persistent volume claim ``pv-blob2`` in another deployment.
+
+# Security consideration
```
Comment (andyzhangx) [inline deploy/example/mountstorage/README.md:271]: let's remove this , make it simple

**Review by andyzhangx — COMMENTED** [review]
/ok-to-test

File: deploy/example/mountstorage/deployment.yaml [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+            mountPath: /usr/share/nginx/html
+      volumes: 
+        - name: pvc-blob1 
+          persistentVolumeClaim: 
+            claimName:  pvc-blob1
+status: {}
+
+---
+
+apiVersion: v1
```
Comment (andyzhangx) [inline deploy/example/mountstorage/deployment.yaml:35]: remove service

File: deploy/example/mountstorage/README.md [hunk 7]
```diff
@@ -0,0 +1,242 @@
...
+        - --file-cache-timeout-in-seconds=120
+      csi:
+        driver: blob.csi.azure.com
+        readOnly: false
+        # make sure this volumeid is unique in the cluster
+        # `#` is not allowed in self defined volumeHandle
+        volumeHandle: pv-blob1
+        volumeAttributes:
+          protocol: fuse
+          resourceGroup: aks-fuseblob-mi
```
Comment (andyzhangx) [inline deploy/example/mountstorage/README.md:92]: rename as `blobfuse-mi`

**Review by andyzhangx — COMMENTED** [review]
mountstorage --> `blobfuse-mi`

and pls also add dynamic provisioning example

File: deploy/example/mountstorage/volume.yaml [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+    - --file-cache-timeout-in-seconds=120
+  csi:
+    driver: blob.csi.azure.com
+    readOnly: false
+    # make sure this volumeid is unique in the cluster
+    # `#` is not allowed in self defined volumeHandle
+    volumeHandle: pv-blob1
+    volumeAttributes:
+      protocol: fuse
+      resourceGroup: aks-fuseblob-mi
```
Comment (andyzhangx) [inline deploy/example/mountstorage/volume.yaml:23]: `blobfuse-mi`

File: deploy/example/mountstorage/README.md [hunk 9]
```diff
@@ -0,0 +1,242 @@
...
+
+- Install the aks-preview Azure CLI extension version 0.5.85 or later.
+
+- Ensure, that you are authenticated or run `az login` 
+
+- Run `az account set --subscription "mysubscription"` to select the right subscription
+
+- Create a storage account container, e.g.
+    ```bash
+    resourcegroup="aks-fuseblob-mi"
```
Comment (andyzhangx) [inline deploy/example/mountstorage/README.md:20]: `blobfuse-mi`

File: deploy/example/mountstorage/README.md [hunk 10]
```diff
@@ -0,0 +1,242 @@
...
+      selector:
+        matchLabels:
+          app: nginx-app1
+      template:
+        metadata:
+          labels:
+            app: nginx-app1
+        spec:
+          containers:
+          - image: mcr.microsoft.com/oss/nginx/nginx:1.19.5
```
Comment (andyzhangx) [inline deploy/example/mountstorage/README.md:133]: use this example config:
https://github.com/kubernetes-sigs/blob-csi-driver/blob/12d5f64c099dd8002ce32c239f522906dbe33506/deploy/example/deployment.yaml#L35-L39

File: deploy/example/mountstorage/README.md (hunk 1 above)
Comment (umagnus) [reply deploy/example/mountstorage/README.md:28]: done

File: deploy/example/mountstorage/README.md (hunk 2 above)
Comment (umagnus) [reply deploy/example/mountstorage/README.md]: done

File: deploy/example/mountstorage/README.md (hunk 3 above)
Comment (umagnus) [reply deploy/example/mountstorage/README.md]: done

File: deploy/example/mountstorage/README.md (hunk 4 above)
Comment (umagnus) [reply deploy/example/mountstorage/README.md]: done

File: deploy/example/mountstorage/README.md (hunk 5 above)
Comment (umagnus) [reply deploy/example/mountstorage/README.md]: done

File: deploy/example/mountstorage/deployment.yaml (hunk 6 above)
Comment (umagnus) [reply deploy/example/mountstorage/deployment.yaml]: done

File: deploy/example/mountstorage/README.md (hunk 7 above)
Comment (umagnus) [reply deploy/example/mountstorage/README.md]: done

File: deploy/example/mountstorage/volume.yaml (hunk 8 above)
Comment (umagnus) [reply deploy/example/mountstorage/volume.yaml:23]: done

File: deploy/example/mountstorage/README.md (hunk 9 above)
Comment (umagnus) [reply deploy/example/mountstorage/README.md]: done

File: deploy/example/mountstorage/README.md (hunk 10 above)
Comment (umagnus) [reply deploy/example/mountstorage/README.md]: done

**Comment by umagnus** [comment]
> mountstorage --> `blobfuse-mi`
> 
> and pls also add dynamic provisioning example

added dynamic provisioning example

File: deploy/example/mountstorage/storageclass-blobfuse-resourcegroup.yaml [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+  name: blob-fuse
+provisioner: blob.csi.azure.com
+parameters:
+  skuName: Premium_LRS 
+  protocol: fuse
+  resourceGroup: EXISTING_RESOURCE_GROUP_NAME
+  storageAccount: EXISTING_STORAGE_ACCOUNT_NAME # optional, if use existing storage account
+  containerName: EXISTING_CONTAINER_NAME # optional, if use existing container
+  AzureStorageAuthType: MSI
+  AzureStorageIdentityClientID: "92926dfd-e61b-4730-85ab-5be73b374e82"
```
Comment (andyzhangx) [inline deploy/example/mountstorage/storageclass-blobfuse-resourcegroup.yaml:14]: use `xxx-xx ..' for this

File: deploy/example/mountstorage/pv-blobfuse-csi-mount.yaml [hunk 12]
```diff
@@ -0,0 +1,27 @@
+apiVersion: v1
```
Comment (andyzhangx) [inline deploy/example/mountstorage/pv-blobfuse-csi-mount.yaml:1]: rename as `deploy/example/blobfuse-mi/pv-blobfuse-mi.yaml`

File: deploy/example/mountstorage/README.md [hunk 13]
```diff
@@ -0,0 +1,253 @@
...
+          containerName: mycontainer
+          AzureStorageAuthType: MSI
+          AzureStorageIdentityClientID: "xxxxx-xxxx-xxx-xxx-xxxxxxx"
+    
+    ---
+    
+    apiVersion: v1
+    kind: PersistentVolumeClaim
+    metadata:
+      name: pvc-blob2
```
Comment (andyzhangx) [inline deploy/example/mountstorage/README.md:233]: don't need this, just use existing `pvc-blob-csi-static.yaml` under example dir

File: deploy/example/mountstorage/README.md [hunk 14]
```diff
@@ -0,0 +1,253 @@
...
+    az identity list -g -g "$resourcegroup" --query "[?name == 'myaksblobmi2'].principalId" -o tsv
+    ```
+
+1. Create a ``volume2.yaml`` file and set objectID for ``AzureStorageIdentityClientID``. \
+   Please also check ``resourceGroup`` and ``storageAccount``.
+    ```yml
+    apiVersion: v1
+    kind: PersistentVolume
+    metadata:
+      name: pv-blob2
```
Comment (andyzhangx) [inline deploy/example/mountstorage/README.md:203]: just use `pv-blobfuse-csi`, then you could reuse existing `pvc-blob-csi-static.yaml`

File: deploy/example/blobfuse-mi/README.md [hunk 15]
```diff
@@ -0,0 +1,233 @@
...
+      name: blob-fuse
+    provisioner: blob.csi.azure.com
+    parameters:
+      skuName: Premium_LRS 
+      protocol: fuse
+      resourceGroup: EXISTING_RESOURCE_GROUP_NAME
+      storageAccount: EXISTING_STORAGE_ACCOUNT_NAME # optional, if use existing storage account
+      containerName: EXISTING_CONTAINER_NAME # optional, if use existing container
+      AzureStorageAuthType: MSI
+      AzureStorageIdentityClientID: "xxxxx-xxxx-xxx-xxx-xxxxxxx"
```
Comment (andyzhangx) [inline deploy/example/blobfuse-mi/README.md:48]: how is this AzureStorageIdentityClientID generated? I think you need to add get ClientID into `Before you begin` section?

Comment (umagnus) [reply deploy/example/blobfuse-mi/README.md:53]: add it

**Review by andyzhangx — APPROVED** [review]
/lgtm

**Pull request merged**